agent-install/agent-install.sh

```diff
@@ -35,1269 +34,1048 @@ CONFIGMAP_NAME="openhorizon-agent-config"
+    # Next get config file values (cmd line was already parsed), so get_variable can apply the precedence order
+    get_variable AGENT_CFG_FILE 'agent-install.cfg'
+    read_config_file "$AGENT_CFG_FILE"
 
-    if [ "${DEPLOY_TYPE}" == "device" ]; then
-    	# preliminary check for script arguments
-    	check_empty "$PKG_PATH" "path to installation packages"
-    	if [[ ${PKG_PATH:0:4} == "http" ]]; then
-	    PKG_APT_REPO="$PKG_PATH"
-	    if [[ "${PKG_APT_REPO: -1}" == "/" ]]; then
-		    PKG_APT_REPO=$(echo "$PKG_APT_REPO" | sed 's/\/$//')
-	    fi
-	    PKG_PATH="."
-    	else
-	    PKG_PATH=$(echo "$PKG_PATH" | sed 's/\/$//')
-	    check_exist d "$PKG_PATH" "The package installation"
-    	fi
+    # Now that we have the values from cmd line and config file, we can get all of the variables
+    get_variable AGENT_VERBOSITY 3
+    # need to check this value right now, because we use it immediately
+    if [[ $AGENT_VERBOSITY -lt 0 || $AGENT_VERBOSITY -gt $VERB_DEBUG ]]; then
+        log_fatal 1 "AGENT_VERBOSITY must be in the range 0 - $VERB_DEBUG"
+    fi
+    get_variable AGENT_SKIP_REGISTRATION 'false'
+    get_variable HZN_EXCHANGE_URL '' 'true'
+    get_variable HZN_FSS_CSSURL '' 'true'
+    get_variable HZN_ORG_ID '' 'true'
+    get_variable HZN_EXCHANGE_NODE_AUTH   #future: maybe these 3 should be combined
+    get_variable NODE_ID
+    get_variable HZN_DEVICE_ID
+    get_variable HZN_EXCHANGE_USER_AUTH
+    get_variable HZN_MGMT_HUB_CERT_PATH
+    get_variable AGENT_CERT_FILE "${HZN_MGMT_HUB_CERT_PATH:-agent-install.crt}"
+    get_variable HZN_EXCHANGE_PATTERN
+    get_variable HZN_NODE_POLICY
+    get_variable AGENT_WAIT_FOR_SERVICE
+    get_variable AGENT_WAIT_FOR_SERVICE_ORG
+    get_variable AGENT_REGISTRATION_TIMEOUT
+    get_variable AGENT_OVERWRITE 'false'
+    get_variable AGENT_SKIP_PROMPT 'false'
+    get_variable AGENT_INSTALL_ZIP 'agent-install-files.tar.gz'
+    get_variable AGENT_DEPLOY_TYPE 'device'
+    get_variable AGENT_WAIT_MAX_SECONDS '30'
+
+    if [[ "${AGENT_DEPLOY_TYPE}" == "device" ]]; then
+        get_variable PKG_PATH '.'
+        get_variable NODE_ID_MAPPING_FILE 'node-id-mapping.csv'
+        get_variable PKG_APT_KEY
+        get_variable APT_REPO_BRANCH 'updates'
+    elif [[ "${AGENT_DEPLOY_TYPE}" == "cluster" ]]; then
+        get_variable EDGE_CLUSTER_STORAGE_CLASS 'gp2'
+        get_variable AGENT_NAMESPACE 'openhorizon-agent'
+        get_variable USE_EDGE_CLUSTER_REGISTRY 'true'
+        get_variable AGENT_DEPLOYMENT_STATUS_TIMEOUT_SECONDS '75'
+
+        if [[ "$USE_EDGE_CLUSTER_REGISTRY" == "true" ]]; then
+            #lily: do we really need the user to input IMAGE_ON_EDGE_CLUSTER_REGISTRY and AGENT_IMAGE_TAG. These seem like internal values (our code creates the image in the local registry with these values and then uses that image). Can we just always default them based on the edge cluster type?
+            get_variable IMAGE_ON_EDGE_CLUSTER_REGISTRY '' 'true'
```

Comment:
`IMAGE_ON_EDGE_CLUSTER_REGISTRY`:
OCP - `<registry-host>/<agent-project>/amd64_anax_k8s`
microk8s - `localhost:32000/<agent-namespace>/amd64_anax_k8s`
k3s - `<docker-registry-service-endpoint>/openhorizon-agent/agent_amd64`

So if we want to default `IMAGE_ON_EDGE_CLUSTER_REGISTRY` for different types, we might need to retrieve edge OCP `<registry-host>` on the fly. Also we need to get the `<docker-registry-service-endpoint>` for k3s

Reply:
The `AGENT_IMAGE_TAG` is actually put into `agent-install.cfg` file by `edgeNodeFiles.sh`. The value is passed in to `edgeNodeFiles.sh` by specifying `-i` (if not, use default image tag). So it is info from the agent image. 
But I think it won't hurt anything if we remove this value from cfg file, and variable check in `agent-install.sh`. If the agent image tag is removed, prob we need to parse the image name and extract the version in install script?

Reply:
Yes, that's what i meant, have the code figure out what the values should be